app/indexers/concerns/curator/indexer/attachment_indexer.rb

```diff
@@ -8,44 +9,42 @@ module AttachmentIndexer
         configure do
           each_record do |record, context|
             attachments = {}
+            attachment_fields = %i(byte_size content_type checksum).freeze
+
             has_one_attachment_types = %i(audio_access audio_master document_access document_master
+                                          ebook_access_epub ebook_access_mobi ebook_access_daisy
                                           image_access_800 image_georectified_master image_master image_negative_master
                                           image_service image_thumbnail_300 metadata_foxml metadata_ia metadata_ia_scan
                                           metadata_marc_xml metadata_mods metadata_oai text_coordinates_access
                                           text_coordinates_master text_plain video_access video_master).freeze
-            has_many_attachment_types = %i(ebook_access).freeze
-            attachment_fields = %i(byte_size content_type checksum).freeze
-
             has_one_attachment_types.each do |attachment_type|
-              next unless record.respond_to?(attachment_type)
+              next unless record.respond_to?(attachment_type) && record.send(attachment_type).attached?
```

Comment:
you could do just `record.send(attachment_type)&.attached?` here instead

Reply:
Tried that, but it returns a `NoMethodError: undefined method` for any attachment_type that isn't defined in the model.